src/mame/sinclair/specnext_sprites.cpp

```diff
@@ -0,0 +1,282 @@
+// license:BSD-3-Clause
+/**********************************************************************
+
+    Spectrum Next Sprites
+
+TODO:
+- Relative Sprite, Composite Type
+- Secondary Palette
+
+**********************************************************************/
+
+#include "emu.h"
+#include "screen.h"
+
+#include "specnext_sprites.h"
+
+static const gfx_layout gfx_16x16x8 =
+{
+    16, 16, RGN_FRAC(1, 1), 8,
+    { STEP8(0, 1) },
+    { STEP16(0, 8) },
+    { STEP16(0, 8 * 16) },
+    8 * 16 * 16
+};
+
+static const gfx_layout gfx_16x16x4 =
+{
+    16, 16, RGN_FRAC(1, 1), 4,
+    { STEP4(0, 1) },
+    { STEP16(0, 4) },
+    { STEP16(0, 4 * 16) },
+    4 * 16 * 16
+};
+
+static const gfx_layout gfx_16x16x8_r =
+{
+    16, 16, RGN_FRAC(1, 1), 8,
+    { STEP8(0, 1) },
+    { STEP16(8 * 16 * 15, -8 * 16) },
+    { STEP16(0, 8) },
+    8 * 16 * 16
+};
+
+static const gfx_layout gfx_16x16x4_r =
+{
+    16, 16, RGN_FRAC(1, 1), 4,
+    { STEP4(0, 1) },
+    { STEP16(4 * 16 * 15, -4 * 16) },
+    { STEP16(0, 4) },
+    4 * 16 * 16
+};
+
+static GFXDECODE_START( gfx_sprites )
+    GFXDECODE_DEVICE_RAM( "pattern_ram", 0, gfx_16x16x8, 0x600, 256 )
+    GFXDECODE_DEVICE_RAM( "pattern_ram", 0, gfx_16x16x4, 0x600, 16 )
+    GFXDECODE_DEVICE_RAM( "pattern_ram", 0, gfx_16x16x8_r, 0x600, 256 )
+    GFXDECODE_DEVICE_RAM( "pattern_ram", 0, gfx_16x16x4_r, 0x600, 16 )
+GFXDECODE_END
+
+specnext_sprites_device::specnext_sprites_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+	: device_t(mconfig, SPECNEXT_SPRITES, tag, owner, clock)
+    , device_gfx_interface(mconfig, *this, gfx_sprites, ":palette")
```

Comment:
Don’t make assumptions about other devices in the system.  Provide a means of configuration if necessary.